### PR TITLE
Create documentation for the __bitXYZ functions. 

### PR DESCRIPTION
**Pull request by irfan-iiitr**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added documentation  __bitXYZ functions. This PR will add docs about  __bitBoolMaskAnd,  __bitBoolMaskOr,__bitSwapLastTwo
, __bitWrapperFunc

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

 __bitBoolMaskOr(mask_a, mask_b) 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by Blargian** [comment]
So far so good. Let's add `__bitBoolMaskAnd`, `__bitSwapLastTwo` and `__bitWrapperFunc`.

**Comment by irfan-iiitr** [comment]
__bitBoolMaskAnd, __bitSwapLastTwo,__bitWrapperFunc added.

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 8b15be44798288a2f0cbbae5fbf3dc57f16dceb3 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66469/8b15be44798288a2f0cbbae5fbf3dc57f16dceb3/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66469/8b15be44798288a2f0cbbae5fbf3dc57f16dceb3/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: docs/en/sql-reference/functions/bit-functions.md [hunk 1]
```diff
@@ -438,3 +438,212 @@ Result:
...
+```response
+┌─id─┬─toTypeName(id)─┬─result─┬─toTypeName(result)─┐
+│  1 │ UInt32         │      3 │ UInt8              │
+│  2 │ UInt32         │      3 │ UInt8              │
+└────┴────────────────┴────────┴────────────────────┘
+```
+
+The example demonstrates how the `__bitBoolMaskAnd` function combines the boolean masks.
+
+# __bitSwapLastTwo Function
```
Comment (Avogar) [inline docs/en/sql-reference/functions/bit-functions.md:557]:
```suggestion
# __bitSwapLastTwo
```

File: docs/en/sql-reference/functions/bit-functions.md [hunk 2]
```diff
@@ -438,3 +438,212 @@ Result:
...
+The expected output for the SQL query provided in the example usage section, demonstrating the `__bitBoolMaskAnd` function's effect on combining boolean masks, is as follows:
+
+```response
+┌─id─┬─toTypeName(id)─┬─result─┬─toTypeName(result)─┐
+│  1 │ UInt32         │      3 │ UInt8              │
+│  2 │ UInt32         │      1 │ UInt8              │
+└────┴────────────────┴────────┴────────────────────┘
+```
+
+## `__bitWrapperFunc`
```
Comment (Avogar) [inline docs/en/sql-reference/functions/bit-functions.md:606]:
```suggestion
## __bitWrapperFunc
```

**Comment by Blargian** [comment]
> BTW, what is the motivation of documenting these functions? They are used only internally during indexes analysis, And the names are started with `__` that means that they are not supposed to be used by users.

Motivation comes purely from working through the list of functions in [#1833](https://github.com/ClickHouse/clickhouse-docs/issues/1833). I'm not sure with which criteria that list was generated. 

Maybe we can group all functions for internal use under a new category `internal functions` or `developer functions`, or something similar?

**Comment by Avogar** [comment]
Probably these functions were added to the list because we have show in `system.functions` table. And I actually think that we don't need to document these functions at all:
- They are supposed to be used only internally during indexes analisys, users should know nothing about it.
- We can reimplement hte analisys and remove these functions any time

**Comment by Blargian** [comment]
> Probably these functions were added to the list because we show them in `system.functions` table. And I actually think that we don't need to document these functions at all:
> 
> * They are supposed to be used only internally during indexes analisys, users should know nothing about it.
> * We can reimplement this analisys and remove these functions any time

Clear. I have closed [1926](https://github.com/ClickHouse/clickhouse-docs/issues/1926) and also [1927](https://github.com/ClickHouse/clickhouse-docs/issues/1927).

**Pull request closed without merging**